docs/root/configuration/http/http_filters/tcp_post_filter.rst

```diff
@@ -0,0 +1,16 @@
+.. _config_http_filters_tcp_post:
+
+TCP Post
+===============
+
+The TcpPost filter is used by the upstream proxy to coordinate with downstream proxy that proxies
+TCP streams over HTTP/2 POST requests. 
+
+Configuration
+-------------
+
+* This filter should be configured with the name *envoy.filters.http.tcp_post*.
+
+It converts POST requests back to CONNECT to trigger the regular TCP decapping. Therefore, it's
+typically used with another localhost listener that handles the regular TCP decapping for CONNECT
```

Comment:
I think a config example here would be worthwhile?

Also, stepping back, I thought at one point we discussed potentially building this directly into the CONNECT handling path in HCM so that we don't have to do a loop back? If we did that we don't need a filter and I think the entire thing is much simpler? Did we discard that idea for some reason?

Reply:
I added a new API to router, which could help avoid adding this http filter, please check the comment at the top, thx!